xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/renderer/XWikiDocumentURITitleGenerator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.internal.renderer;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.PageAttachmentReference;
+import org.xwiki.model.reference.PageAttachmentReferenceResolver;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.rendering.renderer.reference.link.URITitleGenerator;
+
+/**
+ * Generate link titles for ATTACH URIs.
+ *
+ * @version $Id$
+ * @since 15.2RC1
+ */
+@Component(hints = {"doc", "page"})
+@Singleton
+public class XWikiDocumentURITitleGenerator implements URITitleGenerator
+{
+    @Inject
+    private ContextualLocalizationManager contextLocalization;
+
+    /**
+     * Used to extract the attachment name part in an Attachment reference.
+     */
+    @Inject
+    @Named("current")
+    private PageAttachmentReferenceResolver<String> currentAttachmentReferenceResolver;
+
+    @Override
+    public String generateCreateTitle(ResourceReference reference)
+    {
+        PageAttachmentReference attachmentReference =
+            this.currentAttachmentReferenceResolver.resolve(reference.getReference());
+        return this.contextLocalization.getTranslationPlain("core.create.inline.label",
+         attachmentReference.getName());
```

Comment:
I don't understand, you're returning "Create page: attachmentName"? Doesn't make any sense. Also bad formatting.

Reply:
:+1: for the formatting.
When used on a rendered link, it seems to me like it works properly, the reference name is the name of the page we're looking to create. What should I use in its stead?

For reference, I just checked again this code in action:
> [[Dead link>>The dead page that should be created]]

![19383-titleGenerator](https://user-images.githubusercontent.com/28761965/228490471-877c23f7-65f1-4fe9-bbab-5281c4b7aca4.png)